endpoint.go

```diff
@@ -349,44 +249,36 @@ func (ep *endpoint) joinLeaveEnd() {
 	}
 }
 
-func (ep *endpoint) Join(containerID string, options ...EndpointOption) error {
+func (ep *endpoint) Join(sbox Sandbox, options ...EndpointOption) error {
 	var err error
 
-	if containerID == "" {
-		return InvalidContainerIDError(containerID)
+	if sbox == nil {
+		return types.BadRequestErrorf("endpoint cannot be joined by nil container")
+	}
+
+	s, ok := sbox.(*sandboxData)
+	if !ok {
+		return types.BadRequestErrorf("not a valid Sandbox interface")
 	}
```

Comment:
Actually, libnetwork controls the lifecycle of Sandbox and caller of this API can succeed only if it passes a Sandbox which he received from libnetwork.
There should not be any need for the Sandbox presence check.

I'd prefer to add the check if we find the issue is real and reproducible. I don't want to hide other underlying issues with this corner case check for now.